pages/sectionEditor/SubmissionEditHandler.inc.php

```diff
@@ -275,10 +286,214 @@ function submissionReview($args, &$request) {
 		$templateMgr->assign('allowRecommendation', $allowRecommendation);
 		$templateMgr->assign('allowResubmit', $allowResubmit);
 		$templateMgr->assign('allowCopyedit', $allowCopyedit);
+		
+		$templateMgr->assign('isEditor', $isEditor);
+		$templateMgr->assign('draftingDisabled', $draftingDisabled);
+		$templateMgr->assign('drafts', $drafts);
 
 		$templateMgr->assign('helpTopicId', 'editorial.sectionEditorsRole.review');
 		$templateMgr->display('sectionEditor/submissionReview.tpl');
 	}
+	
+	/**
+	 * View the submission review page.
+	 * @param $args array
+	 * @param $request PKPRequest
+	 */
+	function draftDecision($args, &$request) {
+		$articleId = isset($args[0]) ? (int) $args[0] : 0;
+		$this->validate($articleId, SECTION_EDITOR_ACCESS_REVIEW);
+		$journal =& Request::getJournal();
+		$submission =& $this->submission;
+		$user =& $request->getUser();
+		$this->setupTemplate(true, $articleId);
+
+		if ($_SERVER['REQUEST_METHOD'] === 'POST' && isset($_POST['submit_draft'])) {
+
+			$decision = $_POST["decision"];
+			$senioreditorId = $_POST["editor"];
+			$subject = $_POST["subject"];
+			$body = $_POST["body"];
+			$note = $_POST["note"];
+			$status = 'draft';
+			$key = uniqid('draft_', true);
+			$junioreditorId = $user->getId();
+
+			$articleDao =& DAORegistry::getDAO('ArticleDAO');
+			$article =& $articleDao->getArticle($articleId);
+
+			//$articleID, $Key, $seniorEditorID, $juniorEditorID, $Subject, $Body, $Note, $Decision, $Status
+			$sectioneditorSubmissionDao =& DAORegistry::getDAO('SectionEditorSubmissionDAO');
+			$draft = $sectioneditorSubmissionDao->recordDraftDecision($articleId, $key, $senioreditorId, $junioreditorId, $subject, $body, $note, $decision, $status);
+
+			// lets send an email to the editor :D
+			$base_url = Config::getVar('general', 'base_url');
+			$senior_editor_email = $sectioneditorSubmissionDao->getUserEmail($senioreditorId);
+			$email_message = $article->getArticleTitle() . " has a new draft decision. View the draft: " . $base_url . "/index.php/" . $journal->getPath() ."/editor/viewDraftDecision/" . $articleId . "?key=" . $key;
```

Comment:
This English language text needs to be localized; also, URLs should not be assembled directly. `Request::url` should be used instead.